### PR TITLE
feat: add new 'minus' icon and update TableOfContents component

### PR DESCRIPTION
**Pull request by samanthadotdesign**
- Added new minus icon for list item from BoxIcon
- Updated typography and semantic colour tokens
- Changed underline on hover with underline offset

### CODE REVIEW

File: packages/components/src/templates/next/components/internal/TableOfContents/TableOfContents.tsx [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
-      <div className="flex flex-col gap-3">
+    <div className="flex flex-col gap-3">
+      <p className="prose-headline-lg-medium text-base-content-strong">
+        On this page
+      </p>
+      <div className="prose-body-base flex flex-col gap-3">
         {items.map(({ anchorLink, content }) => (
           <div className="flex flex-row items-start gap-2">
-            <BiRightArrowAlt className="size-6 shrink-0 text-interaction-main" />
+            <BiMinus className="size-5 shrink-0 fill-base-content-strong" />
```
Comment (dcshzj) [inline packages/components/src/templates/next/components/internal/TableOfContents/TableOfContents.tsx:14]: issue: I think the minus sign is not middle aligned
Screenshot 2024-07-29 at 11 18 59

File: packages/components/src/common/icons.ts [hunk 2]
```diff
@@ -31,4 +33,5 @@ export const SUPPORTED_ICONS_MAP: Record<SupportedIconName, SupportedIconType> =
     "office-building": BiBuildings,
     stars: BiStar,
     globe: BiGlobe,
+    minus: BiMinus,
```
Comment (dcshzj) [inline packages/components/src/common/icons.ts:36]: question: hmm the icon here means that we are making it available for users to use this icon (for instance in infocols), is this intended?

**Review by dcshzj — COMMENTED** [review]
Personal opinion but I thought having the vertical bar was nice HAHA but yeah defer to your professional judgement :)

File: packages/components/src/common/icons.ts (hunk 2 above)
Comment (sehyunidaaa) [reply packages/components/src/common/icons.ts]: removed in https://github.com/opengovsg/isomer/pull/377/commits/b6a81d55cccf307d154baf1d4ed64fa390f22e95

File: packages/components/src/templates/next/components/internal/TableOfContents/TableOfContents.tsx (hunk 1 above)
Comment (sehyunidaaa) [reply packages/components/src/templates/next/components/internal/TableOfContents/TableOfContents.tsx]: addressed in https://github.com/opengovsg/isomer/pull/377/commits/26b423430ba1c7aeb875ebcb8c830ef7b79eb434

**Review by dcshzj — APPROVED** [review]

**Pull request merged**